en/simulation/failsafes.md

```diff
@@ -51,7 +48,25 @@ To control how fast the battery depletes to the minimal value use the parameter
 By changing [SIM_BAT_MIN_PCT](../advanced_config/parameter_reference.md#SIM_BAT_MIN_PCT) in flight, you can also test regaining capacity to simulate inaccurate battery state estimation or in-air charging technology.
 :::
 
-## GPS Loss
-
-To simulate losing and regaining GPS information you can just stop the publication of GPS messages.
-This is done by running the `param set SIM_GPS_BLOCK 1` and `param set SIM_GPS_BLOCK 0` commands on your SITL instance *pxh shell* to block and unblock messages respectively.
+## Sensor/System Failure
+
+System failure injection can be used to simulate different types of failures in many sensors and systems (e.g. GPS, barometer, gyro, avoidance etc.):
```

Comment:
@julianoes Several questions about Failure injection:
1. Is this just for simulation, or can it be used on a real vehicle in flight? My assumption is yes, and that is why we have the `SYS_FAILURE_EN` param (to make sure this doenn't happen by accident)
1. Is the `SYS_FAILURE_EN` enabled by default for simulation and/or HITL?
1. Assuming it can be used in normal flight when would you do this?
1. What are the different failure types. I.e. I get "on and off" but what is the difference between stuck, off and intermittent. What is the difference between wrong and garbage. Basically - when should someone use each?

If this is intended to be generic my thinking is I might move this out of simulation into the Debugging/Logging section and link to this from here.

Reply:
1. Exactly, it's for both, simulation and real, hence the param.
2. The failure module and therefore param is not built for "real" yet, so it's not available yet. But I should do that soon.
3. To test the behaviour, e.g. to test the RC loss behaviour in flight (without having to switch off RC).
4. Stuck is stuck on a certain value but it keeps publishing it. This could be a malfunctioning sensor.
Off means the sensor does not publish anymore, so it can be detected more easily.
Intermittent is, as the name implies, intermittent, switch between off and on.
Wrong would be a sensor reporting values that are wrong but they might still "look correct". Garbage is just random noice, like reading uninitialized memory.
Most of these are not implemented broadly yet, so they will likely return "Unsupported" when you try them.

Reply:
And you're right it could go in a separate page once it has a bit better support.

Thanks for updating it here though, that's great!

Reply:
Awesome. You'll be even happier it now has its own page: https://github.com/PX4/PX4-user_guide/pull/1351